pkg/registry/apis/secret/accesscontrol_test.go

```diff
@@ -0,0 +1,160 @@
+package secret
+
+import (
+	"context"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/services/accesscontrol/acimpl"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+func TestSecretAuthorizer(t *testing.T) {
+	t.Run("when the request is not for a resource, the decision is 'no opinion'", func(t *testing.T) {
+		ctx := context.Background()
+
+		sa := SecretAuthorizer(nil)
+
+		attr := &fakeAttributes{isResource: false}
+
+		decision, reason, err := sa.Authorize(ctx, attr)
+		require.Equal(t, authorizer.DecisionNoOpinion, decision)
+		require.Empty(t, reason)
+		require.NoError(t, err)
+	})
+
+	t.Run("when the request verb is not Create or List and the name is empty, the decision is 'deny'", func(t *testing.T) {
```

Comment:
Just for completeness, we could run this test with every verb except for create and list. Maybe the request should include the namespace so the test is different from the next one?

Reply:
💯 I will continue the tests and cover these scenarios, just wanted to get the PR out fast for feedback on the implementation part.